opal/mca/base/mca_base_var_enum.c

```diff
@@ -117,6 +117,96 @@ mca_base_var_enum_t mca_base_var_enum_bool = {
     .dump      = mca_base_var_enum_bool_dump
 };
 
+static int mca_base_var_enum_auto_bool_get_count (mca_base_var_enum_t *enumerator, int *count)
+{
+    *count = 3;
+    return OPAL_SUCCESS;
+}
+
+static int mca_base_var_enum_auto_bool_get_value (mca_base_var_enum_t *self, int index,
+                                                  int *value, const char **string_value)
+{
+    const int values[3] = {0, 1, -1};
+    const char *strings[3] = {"false", "true", "auto"};
+
+    if (2 < index) {
+        return OPAL_ERR_VALUE_OUT_OF_BOUNDS;
+    }
+
+    *value = values[index];
+    *string_value = strings[index];
+
+    return OPAL_SUCCESS;
+}
+
+static int mca_base_var_enum_auto_bool_vfs (mca_base_var_enum_t *self, const char *string_value,
+                                            int *value)
+{
+    char *tmp;
+    int v;
+
+    /* skip whitespace */
+    string_value += strspn (string_value, " \t\n\v\f\r");
```

Comment:
If you trim excess whitespace from the front, shouldn't you also trim excess whitespace from the back?  I.e., `___true`, will be accepted, but `___true____` will not (substitute ` ` for `_` -- I couldn't get markdown to render trailing spaces).